transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionEntity.java

```diff
@@ -133,6 +135,32 @@ private Behavior paymentPending(TransactionState state) {
     private Behavior paymentSubmitted(TransactionState state) {
         BehaviorBuilder builder = newBehaviorBuilder(state);
 
+        builder.setCommandHandler(SubmitPaymentStatus.class, (cmd, ctx) -> {
+            if (cmd.getUserId().equals(state().getTransaction().get().getCreator())) {
+                if(cmd.isApproved())
+                    return ctx.thenPersist(new PaymentApproved(entityUUID()), (e) -> ctx.reply(Done.getInstance()));
+                else
+                    return ctx.thenPersist(new PaymentRejected(entityUUID()), (e) -> ctx.reply(Done.getInstance()));
+            } else
+                throw new Forbidden("Only the item creator can approve or reject payment");
+        });
+
+        builder.setEventHandlerChangingBehavior(PaymentApproved.class, evt ->
+                paymentConfirmed(state().withStatus(TransactionStatus.PAYMENT_CONFIRMED))
+        );
+
+        builder.setEventHandlerChangingBehavior(PaymentRejected.class, evt ->
+                paymentConfirmed(state().withStatus(TransactionStatus.PAYMENT_PENDING))
+        );
```

Comment:
I reviewed the class as a whole and I think the method should be `transactionFinished` and the status `PAYMENT_CONFIRMED` and `PAYMENT_REJECTED`. I'm avoiding words like `confirmed` or `completed` that give a sense of success. IIUC, this is a transition into a final state that users can't get out of and whic contains a status of success or failure.